starcounter-include.html

```diff
@@ -210,35 +208,67 @@
          */
         StarcounterIncludePrototype._compositionChanged = function(compositionString) {
-            let composition;
-            if (this.partial && compositionString === undefined) {
-                const compositionProvider = findCompositionProvider(this.partial);
-                if (compositionProvider) {
-                    compositionString = compositionProvider.Composition$; //should always be string
-                    this.storedLayout = compositionString;
-                }
-            }
-            // do not change if there is one, it's same as before, and we are not forced
-            if (!compositionString) {
-                composition = this.defaultComposition && this.defaultComposition.cloneNode(true);
-                compositionString = this.defaultComposition;
-            } else {
-                // make clone to avoid direct binding
-                composition = this.stringToDocumentFragment(compositionString);
-            }
-            if (!compositionString || compositionString !== this._lastLayout || this._forceLayoutChange) {
-                this.stampComposition(composition);
-                this._forceLayoutChange = false;
-                this._lastLayout = compositionString;
+            var that = this;
+
+            if (compositionString) {
+                this._forceLayoutChange = true;
+                this._renderCompositionChange("explicit or fallback", function() {
+                    return that.stringToDocumentFragment(compositionString);
+                });
             }
+            else {
+                const compositionProvider = this.partial && findCompositionProvider(this.partial);
+                this.partialId = compositionProvider && compositionProvider.PartialId;
+
+                if (compositionProvider && compositionProvider.Composition$ && compositionString === "") {
+                    //this is a request from starcounter-layout-html-editor to reset to default composition
+                    that.Composition$ = "";
+                }
+
+                if (compositionProvider && compositionProvider.Composition$) {
+                    this._renderCompositionChange(compositionProvider.Composition$, function() {
+                        that.storedLayout = compositionProvider.Composition$; //should always be string
+                        return that.stringToDocumentFragment(that.storedLayout);
+                    });
+                }
+                else if (this.defaultComposition) {
+                    this._renderCompositionChange(this.defaultComposition, function() {
+                        return that.defaultComposition.cloneNode(true);
+                    });
+                }
+                else {
+                    this._renderCompositionChange("explicit or fallback", function() {
+                        return undefined;
+                    });
+                }
+            }            
```

Comment:
in `_compositionChanged `, arrow functions would: 
1. Look way nicer
2. Diminish the need for `that`.